### DISCUSSION TITLE
C++ and OCPP

### DISCUSSION THREAD

**Discussion by clagare** [community/community#122050] [Programming Help]
### Body

From my Visual C++ program, I want to control, start and stop EV charging. It should work for all cars that use OCPP. I can't find how to link the OCPP protocol API. Nor can I find any documentation of the function calls.

Maybe no one has done this for GitHub yet?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by CopperEagle** [comment]
Hey @clagare 

Admittedly, OCPP is unfamiliar territory for me. Does [libocpp](https://github.com/EVerest/libocpp) cover your needs? The repository is in C++, active and fallows the developments in supporting OCPP 1.6 and 2.0.1. It was also "tested against the OCPP Compliance Test Tool (OCTT and OCTT2) during the implementation" and "[various] CSMS", according to the repository. For the latter, it has listed which ones.

**Reply by clagare** [reply]
It was good advice to try libocpp. Reading through all the text is very extensive, so I try to take a shortcut, such as getting examples.

When I study the text in libocpp it says, for example:
•	register_pause_charging_callback
this callback is used by libocpp to request pausing of charging, the "connector" parameter tells you which connector/EVSE has to pause charging

But how is a function call made? libocpp also includes a link to the OPEN CHARGE POINT PROTOCOL. Here you can ask questions, which I will do in due course.

There may be another possibility via curl, but again I need examples.

The question I have is independent of which programming language you use, so more people should be interested in this.

I will return when I get more facts.
